### PR TITLE
ECR-1551 Add read request to the cryptocurrency service API

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview

Add GET request to the cryptocurrency service API and refactor ApiController to some extent, similar to QaService controller.

---
See: https://jira.bf.local/browse/ECR-1551

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated tests
- [x] The [coding guidelines](https://google.github.io/styleguide/javaguide.html) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [ ] The [continuous integration build](https://www.travis-ci.com/exonum/exonum-java-binding) passes

### CODE REVIEW

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java [hunk 1]
```diff
@@ -62,18 +62,18 @@ public HashCode submitTransaction(Transaction tx) {
...
       MapIndex<PublicKey, Wallet> wallets = schema.wallets();
-      if (!wallets.containsKey(walletId)) {
+      if (!wallets.containsKey(ownerKey)) {
         return Optional.empty();
       }
 
-      Wallet value = wallets.get(walletId);
-      return Optional.of(new Wallet(value.getBalance()));
+      Wallet value = wallets.get(ownerKey);
+      return Optional.of(value);
```
Comment (dmitry-timofeev) [inline exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java]: I believe these lines might be compressed into 
```java
Optional.ofNullable(wallets.get(ownerKey));
```

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyService.java [hunk 2]
```diff
@@ -11,5 +11,5 @@
 
   HashCode submitTransaction(Transaction tx);
 
-  Optional<Wallet> getValue(PublicKey walletId);
+  Optional<Wallet> getValue(PublicKey ownerKey);
```
Comment (dmitry-timofeev) [inline exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyService.java]: getWallet?

File: exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java [hunk 3]
```diff
@@ -48,11 +50,30 @@ public void createPublicApiHandlers(Node node, Router router) {
...
+
+  @Override
+  @SuppressWarnings("ConstantConditions")
+  public Optional<Wallet> getWallet(PublicKey ownerKey) {
+    checkBlockchainInitialized();
+
+    return node.withSnapshot((view) -> {
+      CryptocurrencySchema schema = new CryptocurrencySchema(view);
+      MapIndex<PublicKey, Wallet> wallets = schema.wallets();
+      if (!wallets.containsKey(ownerKey)) {
```
Comment (dmitry-timofeev) [inline exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java]: These three lines are redundant with `Optional.ofNullable` below — the MapIndex is specified to return null if there is no mapping for the given key.

Comment (MakarovS) [reply exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java]: True! Pardon my obliviousness in this case.

**Comment by dmitry-timofeev** [comment]
I wonder why Circle CI started to build this branch — there is no configuration file :thinking:

**Pull request merged**